staging/operator-registry/alpha/action/migrations/migrations.go

```diff
@@ -0,0 +1,109 @@
+package migrations
+
+import (
+	"fmt"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+type MigrationToken string
+
+const (
+	invalidMigration string = ""
+	NoMigrations     string = "none"
+	AllMigrations    string = "all"
+)
+
+type Migration interface {
+	Token() MigrationToken
+	Help() string
+	Migrate(*declcfg.DeclarativeConfig) error
+}
+
+func newMigration(token string, help string, fn func(config *declcfg.DeclarativeConfig) error) Migration {
+	return &simpleMigration{token: MigrationToken(token), help: help, fn: fn}
+}
+
+type simpleMigration struct {
+	token MigrationToken
+	help  string
+	fn    func(*declcfg.DeclarativeConfig) error
+}
+
+func (s simpleMigration) Token() MigrationToken {
+	return s.token
+}
+
+func (s simpleMigration) Migrate(config *declcfg.DeclarativeConfig) error {
+	return s.fn(config)
+}
+
+func (s simpleMigration) Help() string {
+	return s.help
+}
+
+type Migrations struct {
+	Migrations []Migration
+}
+
+// CloneAllMigrations returns a shallow copy of allMigrations
+// since slices.Clone is not available in the current golang version
+func CloneAllMigrations() []Migration {
+	return append(allMigrations[:0:0], allMigrations...)
+}
```

Comment:
This is a semi-kludge to replace `slices.Clone()` from later golang versions. Hopefully not too odious.